addon/unified-resolver.js

```diff
@@ -63,13 +64,9 @@ const Resolver = DefaultResolver.extend({
     return expandedPath;
   },
 
-  resolve(lookupString) {
-    return this._resolverCache[lookupString] || (this._resolverCache[lookupString] = this._resolve(lookupString));
```

Comment:
The caching uses `lookupString` as the key- since there is also now a `source` to be considered the key for a cache would need to be a composite of those two things, not just the one.

But a composite key also means generating a new string on every lookup. My suggestion is to drop the caching at this level for now, especially in light of the Glimmer lookup potentially being static. We can try to re-introduce it at the appropriate level later on. Perhaps even at drastically different levels.

For example when looking up a component both the component and the template could be stored as a cached value together, instead of a cached `resolve` happening for each item. However for services, only the single thing (service) would need to be a cached value.